src/main/java/org/pawaneuler/DataTypes/Rule/Frequency.java

```diff
@@ -0,0 +1,100 @@
+package org.pawaneuler.DataTypes.Rule;
+
+import java.util.ArrayList;
+
+import org.pawaneuler.DataTypes.Trie.Node;
+import org.pawaneuler.DataTypes.Trie.Trie;
+
+public class Frequency {
+    private Trie T;
+    private ArrayList<String> itemset;
+    private int allFreq;
+
+    public Frequency(Trie T) {
+        this.T = T;
+    }
+
+    public int generateAllFreq() {
+        ArrayList<Node> nodes = T.getNodes();
+        int sum = 0;
+        for (int i = 0; i < nodes.size(); i++) {
+            sum += T.getNodeAt(i).getFrequency();
+        }
+        return sum;
+    }
```

Comment:
Shouldn't it be a responsibility of the class to make the time it creates the object of this class? I think it will be better to make it that way than making the user of this class call this function.

Suggestion : make a factory constructor